python/triton/compiler/code_generator.py

```diff
@@ -1300,8 +1300,15 @@ def visit_BoolOp(self, node: ast.BoolOp):
                 # expression so we do not append it to nontrivial_values.
             else:
                 if value.type.is_block():
-                    warnings.warn(
-                        "Logical operators 'and' and 'or' are deprecated for non-scalar tensors; please use '&' or '|' instead"
+                    lineno = getattr(node, "lineno", None)
```

Comment:
I think we should generalize it for all warnings like what we did for `CompilationError` to print messages in a formatted way

Reply:
totally agree